### PR TITLE
Add new async method for loading users with particular userIds

### PR DESCRIPTION
**Pull request by morozkin**
Signed-off-by: Denis Morozov dmorozkn@gmail.com

### CODE REVIEW

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 1]
```diff
@@ -1201,6 +1201,54 @@ - (void)loadUsers
...
+                }
+            }
+            @catch (NSException *exception)
+            {
+                NSLog(@"[MXFileStore] Warning: MXFileRoomStore file for users group %@ has been corrupted", group);
+            }
+        }
+    }
+    
+    return [loadedUsers copy];
```
Comment (manuroe) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: Why do you return a copy?

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 2]
```diff
@@ -1403,6 +1451,13 @@ - (void)asyncUsers:(void (^)(NSArray<MXUser *> * _Nonnull))success failure:(null
     });
 }
 
+- (void)asyncUsersWithUserIds:(NSArray<NSString *> *)userIds success:(void (^)(NSArray<MXUser *> *users))success failure:(nullable void (^)(NSError * _Nonnull))failure
+{
+    dispatch_async(dispatchQueue, ^{
+        success([self loadUsersWithUserIds:userIds]);
```
Comment (manuroe) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: I missed it in the previous PR for async things but call the ```success``` block on the main thread.

There is a kind of contract in the sdk where:
- API must be called from the main thread (unfortunately not always true)
- and the callback block are called on the main thread

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 3]
```diff
@@ -1403,6 +1451,13 @@ - (void)asyncUsers:(void (^)(NSArray<MXUser *> * _Nonnull))success failure:(null
     });
```
Comment (manuroe) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: Call the success block on the main thread (see my longest comment).

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m [hunk 4]
```diff
@@ -1403,6 +1451,13 @@ - (void)asyncUsers:(void (^)(NSArray<MXUser *> * _Nonnull))success failure:(null
...
+- (void)asyncUsersWithUserIds:(NSArray<NSString *> *)userIds success:(void (^)(NSArray<MXUser *> *users))success failure:(nullable void (^)(NSError * _Nonnull))failure
+{
+    dispatch_async(dispatchQueue, ^{
+        success([self loadUsersWithUserIds:userIds]);
+    });
+}
+
 - (void)asyncRoomsSummaries:(void (^)(NSArray<MXRoomSummary *> * _Nonnull))success failure:(nullable void (^)(NSError * _Nonnull))failure
 {
     dispatch_async(dispatchQueue, ^{
```
Comment (manuroe) [inline MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: Call the success block on the main thread (see my longest comment).

File: MatrixSDK/Data/Store/MXFileStore/MXFileStore.m (hunk 1 above)
Comment (morozkin) [reply MatrixSDK/Data/Store/MXFileStore/MXFileStore.m]: @manuroe  `loadedUsers` is a mutable object and we can send mutable messages even if variable points to immutable object. That is why I return immutable object

**Pull request merged**